pkg/config/nodetreemodel/config.go

```diff
@@ -348,12 +360,16 @@ func (c *ntmConfig) UnsetForSource(key string, source model.Source) {
 	}
 
 	c.sequenceID++
-	receivers := slices.Clone(c.notificationReceivers)
-
-	// notifying all receiver about the updated setting
-	for _, receiver := range receivers {
-		receiver(key, previousValue, newValue, c.sequenceID)
+	notification := model.ConfigChangeNotification{
+		Key:           key,
+		PreviousValue: previousValue,
+		NewValue:      newValue,
+		SequenceID:    c.sequenceID,
+		Receivers:     slices.Clone(c.notificationReceivers),
 	}
+
+	c.notificationChannel <- notification
```

Comment:
One thing that might make this a little cleaner is if we encapsulated this in a method on the type that only bothered sending the notification if `c.notificationReceivers` isn't empty.